src/DataRenderer.tsx

```diff
@@ -59,28 +65,48 @@ function ExpandableObject({
     if (!shouldExpandNodeCalledRef.current) {
       shouldExpandNodeCalledRef.current = true;
     } else {
-      setExpanded(shouldExpandNode(level, value, field));
+      const shouldExpand = shouldExpandNode(level, value, field)
+
+      if (shouldExpand) {
+        onNodeExpanded?.(level, value, field);
+      } else {
+        onNodeCollapsed?.(level, value, field);
+      }
+
+      setExpanded(shouldExpand);
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [shouldExpandNode]);
+  }, [shouldExpandNode, onNodeExpanded, onNodeCollapsed]);
```

Comment:
Seems like if we add these two properties to the dependencies, callbacks will be executed each time the property was updated even user haven't done anything. Also if onNodeExpanded or onNodeCollapsed is anonymous function value will be different for each time we render the component.

Reply:
good point. i'm used to wrapping my callbacks in `useCallback` so this is habit but can't guarantee users will do so